src/canvas.class.js

```diff
@@ -1477,8 +1477,21 @@
         this.fire('selection:cleared', { target: obj });
         obj.fire('deselected');
       }
-      if (this._hoveredTarget === obj) {
-        this._hoveredTarget = null;
+      // TODO: need a way to cleanly check if obj is one of _hoveredTargetN and de-ref it
+      // if (obj === this._hoveredTarget){
+      //   this._hoveredTarget = null;
+      // }
+      // is there a more sane way than looping through ALL properties of *this* ?
+      // i wanted to put them into a ._hoveredTargets[Array]
+      // but, that complicates the logic of fireSyntheticInOutEvents
```

Comment:
strict equality has to work, or we did something wrong somewhere.

Reply:
yeah, i understand the strict equality check, i was just wondering about when i have to search Canvas for all of the properties prefixed with "_hoveredTarget" to search through `_hoveredTarget0`...`_hoveredTargetN` to find a match

my gut instinct was to put ._hoveredTargets in an array that would be easier to loop thorugh, rather than all of `Object.keys(this)`

but then it would require a significant change to how `fireSyntheticInOutEvents` so, i kept it simple for now, even though it's probably not the most efficient way to check for target matches